### ISSUE TITLE
Allow limiting semantic token highlighting to identifiers

### ISSUE THREAD

**Issue by Vexu** [zigtools/zls#338]
Semantic token highlighting is nice and all but using a syntax file like the one provided by [vscode-zig](https://github.com/ziglang/vscode-zig) gives much more control. Limiting semantic token highlighting to only identifiers gives the best of both worlds. I think the C/C++ language server does something similar since the result looks much more like my what I was able to achieve with a quick patch.

<details>
<summary>Zls master version</summary>
<img src="https://user-images.githubusercontent.com/15308111/120354046-d5439f00-c30a-11eb-9a24-2bcf27821618.png"></img>
</details>

<details>
<summary>With patch applied</summary>
<img src="https://user-images.githubusercontent.com/15308111/120354053-d674cc00-c30a-11eb-87fb-db5ab16b6d59.png"></img>
</details>

<details>
<summary>Quick patch to enable this behavior, not sure how correct this is</summary>

```diff
diff --git a/src/semantic_tokens.zig b/src/semantic_tokens.zig
index eae2a54..8c813f3 100644
--- a/src/semantic_tokens.zig
+++ b/src/semantic_tokens.zig
@@ -66,6 +66,18 @@ const Builder = struct {
     }
 
     fn add(self: *Builder, token: ast.TokenIndex, token_type: TokenType, token_modifiers: TokenModifiers) !void {
+        switch (token_type) {
+            .type,
+            .parameter,
+            .variable,
+            .enumMember,
+            .field,
+            .errorTag,
+            .function,
+            .label, => {},
+            else => return,
+        }
+
         const tree = self.handle.tree;
         const starts = tree.tokens.items(.start);
         const tags = tree.tokens.items(.tag);
@@ -159,6 +171,18 @@ const Builder = struct {
     }
 
     fn addDirect(self: *Builder, tok_type: TokenType, tok_mod: TokenModifiers, start: usize, length: usize) !void {
+        switch (tok_type) {
+            .type,
+            .parameter,
+            .variable,
+            .enumMember,
+            .field,
+            .errorTag,
+            .function,
+            .label, => {},
+            else => return,
+        }
+
         const delta = offsets.tokenRelativeLocation(
             self.handle.tree,
             self.previous_position,
```
</details>

**Comment by SuperAuguste** [comment]
Quite the interesting idea, perhaps worthy of a configuration option :)

**Comment by Techatrix** [comment]
implemented by #1064
